src/Formats/FormatFactory.h

```diff
@@ -288,7 +290,17 @@ class FormatFactory final : private boost::noncopyable, public IHints<2>
     void checkFormatName(const String & name) const;
     bool exists(const String & name) const;
 
+    FileBucketInfoPtr createFromBuckets(const String & format, const std::vector<size_t> & buckets);
```

Comment:
This method is very strange, we should remove it. Splitters need to create corresponding `FIleBucketInfo` inside `splitToBuckets`